src/deadline/job_attachments/_utils.py

```diff
@@ -98,3 +99,38 @@ def _is_windows_file_path_limit() -> bool:
 
         return bool(ntdll.RtlAreLongPathsEnabled())
     return True
+
+
+def _glob_paths(
+    path: str, include: List[str] = ["**/*"], exclude: Optional[List[str]] = None
+) -> list[str]:
+    """
+    Glob routine that supports Unix style pathname pattern expansion for includes and excludes.
+    path: Root path to glob.
+    include: Optional, pattern syntax for files to include.
+    exlucde: Optional, pattern syntax for files to exclude.
+    """
+    include_files: List[str | None] = []
+    for input_glob in include:
+        include_files.extend(
+            [
+                path.as_posix() if path.is_file() else None
+                for path in pathlib.Path(path).glob(input_glob)
+            ]
+        )
+    include_files = list(filter(None, include_files))
+
+    if exclude:
+        exclude_files = []
+        for exclude_glob in exclude:
+            exclude_files.extend(
+                [
+                    path.as_posix() if path.is_file() else None
+                    for path in pathlib.Path(path).glob(exclude_glob)
+                ]
+            )
+
+        # use sets to remove duplicates and remove all excluded file.
+        exclude_files = list(filter(None, exclude_files))
+        include_files = list(set(include_files) - set(exclude_files))
```

Comment:
I don't see we have input validation for include_files and exclude_files, what if there are overlap from customer input? From these lines exclude would overrule include, is that expected?

Reply:
Yup expected. Customers get what they configure! 

If you configure `.gitignore` incorrectly, the same would happen too.